tests/test_util.py

```diff
@@ -0,0 +1,71 @@
+"""Test cases for util Module."""
+
```

Comment:
All of this looks good. I think eventually we can move all of these test cases to use a parameterized structure like in [this example](https://www.tutorialspoint.com/pytest/pytest_parameterizing_tests.htm). But for now, I think it's fine since we want to get this PR merged soon.